### PR TITLE
feat: add a sample project for Flink User-defined functions

### PR DESCRIPTION
**Pull request by AdrianVasiliu**
Add a sample project for implementing Flink User-defined functions that can be used in the SQL exported from the Event Processing UI.

### CODE REVIEW

File: event-processing/flink-examples/udf/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+ARG UDF_JAR
+ARG FLINK_IMAGE
+FROM --platform=linux/amd64 ${FLINK_IMAGE}
```
Comment (dibbles) [inline event-processing/flink-examples/udf/Dockerfile:3]: The CR examples are generally updated each release to stay current and if you are not intending to do the same with this then I really think it is worth ensuring the readme 
clearly indicates the the platform and version of flink for which this sample was intended/tested.

Here you are specifically targeting linux/amd64 and in docker-compose.yaml you have a hard coded license value.

**Comment by vigneshibmpub** [comment]
@AdrianVasiliu  Why are the files under `event-processing/flink-examples/udf` folder ?
Shouldn't it be under the `flink/udf` folder?

**Comment by AdrianVasiliu** [comment]
@vigneshibmpub 
> Why are the files under event-processing/flink-examples/udf folder ?
> Shouldn't it be under the flink/udf folder?

1. Because this is nicely consistent with the existing `cr-examples` sibling directory (https://github.com/IBM/ibm-event-automation/tree/main/event-processing/cr-examples/eventprocessing)
2. Because this is what Mark N. indicated.

File: event-processing/flink-examples/udf/Dockerfile (hunk 1 above)
Comment (AdrianVasiliu) [reply event-processing/flink-examples/udf/Dockerfile:3]: @dibbles Yes, the goal is that we don't need to update this at each release. 
>  I really think it is worth ensuring the readme clearly indicates the the platform and version of flink

For the Flink version, the README already provides the detailed steps for users to leverage the Flink image of their IBM Operator for Apache flink install. See step https://github.com/IBM/ibm-event-automation/blob/f0c7254ab8c16a77cabd5e1583b982f1b6e7f62f/event-processing/flink-examples/udf/README.md#procedure (step 1.ii.a).

For the platform, I agree we could mention that in the readme. I'll adjust.

Regarding the hardcoded licence: this is the non-production one, which fits the use case of this docker-compose file. It's for users to test the sample UDF project, possibly with their custom modifications of the code. It's not a production Flink instance by definition...

**Review by dibbles — APPROVED** [review]
My concern is addressed

**Pull request merged**